Task_2/Task_2.cpp

```diff
@@ -1,20 +1,46 @@
-﻿// Task_2.cpp : Этот файл содержит функцию "main". Здесь начинается и заканчивается выполнение программы.
-//
+﻿#include <iostream>
 
-#include <iostream>
+using namespace std;
+
+//2. ЕДИНИЦЫ. Найдите все натуральные числа до m, двоичная запись которых состоит из единиц
+
+void BinaryOne(int upTo);
 
 int main()
 {
-    std::cout << "Hello World!\n";
+	int m = 0;
+
+	cout << "Enter up to what natural number m to output numbers whose binary notation consists of ones: ";
+	cin >> m;
+	cout << endl;
+
+	if (m <= 0)
+	{
+		cout << "Error! m <= 0." << endl;
+
+		return -1;
+	}
+	else {}
+
+	BinaryOne(m);
+
+	return 0;
 }
 
-// Запуск программы: CTRL+F5 или меню "Отладка" > "Запуск без отладки"
-// Отладка программы: F5 или меню "Отладка" > "Запустить отладку"
+void BinaryOne(int upTo)
+{
+	int numbers = 3, degree = 2;
+
+	cout << "Natural numbers up to " << upTo << ", whose binary notation consists of ones: 1";
+
+	while (numbers <= upTo)
+	{
+		cout << ", " << numbers;
+
+		numbers += pow(2, degree);
+
+		degree++;
+	}
```

Comment:
Two actions in your function

Reply:
?

Reply:
![-s15xWudjFU](https://user-images.githubusercontent.com/71211228/111071687-5ba8cb80-84e8-11eb-9527-ddaae0355381.jpg)

Reply:
cout + while = two actions